packages/wdio-cli/src/launcher.ts

```diff
@@ -238,6 +239,24 @@ class Launcher {
         })
     }
 
+    /**
+     * Format the specs into an array of objects with files and retries
+     */
+    formatSpecs(capabilities: (Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities | Capabilities.RemoteCapabilities), specFileRetries: number) {
+        let files: (string | string[])[] = []
+        let returnValue: WorkerSpecs[] = []
+
+        files = this.configParser.getSpecs((capabilities as Capabilities.DesiredCapabilities).specs, (capabilities as Capabilities.DesiredCapabilities).exclude)
+        files.forEach(file => {
+            if (typeof file === 'string') {
```

Comment:
Sorry, new to Typescript.  Since we are dealing with an array of strings or an array of string arrays are we not guaranteed to always be getting either a string or an array? I don't know what runtime type checking there is on user data.  I would normally always implement the drop through to catch anything else, but omitted it here as overkill.  In the next push, I have implemented such a drop through.

Reply:
For type checking I now have to return an empty entry, it would be good we could drop anything that is not a string or an array - please see the new code when I push it.

Reply:
@erwinheitzman Can we close out this conversation ?